### DISCUSSION TITLE
Enable/disable logging on a per alarm basis

### DISCUSSION THREAD

**Discussion by jacquelinegarrahan** [ControlSystemStudio/phoebus#1864] [Ideas]
During our requirements gathering, many of our engineers underscored the importance of configurable alarm logging on a per PV basis. This feature would extend the configuration schema to include a field for “logging_enabled” and require the alarm logging service to monitor the configuration topic for changes to the logging enable/disable field.

I had thought I’d seen this request in a previous issue, but can’t find the reference at present…

**Comment by shroffk** [comment]
Hello,
What are the use cases where we would like to only disable the logging of an alarm and not disable the alarm?

One of the motivations of the alarm logger is that it provides a complete history of the alarms and the actions surrounding it, this would be useful for both auditing and port mortem analysis of events. By allowing disabling of logging we include the possibility of the alarm log being incomplete and potentially misleading.

**Reply by jacquelinegarrahan** [reply]
Hi Kunal,

This was a remnant of the Alarm Handler masks and an attempt to capture all functionality of the Alarm Handler with our new solution. I expect we'll drop this requirement based on continued conversations... I agree with your points
